tests/test_flatten.py

```diff
@@ -0,0 +1,39 @@
+from geopandas import GeoDataFrame
+from pandas import DataFrame
+from pytest import raises
+from shapely.geometry import Point
+
+from crossfire import NestedColumnError, flatten
+
+DICT_DATA = [
+    {
+        "answer": 42,
+        "contextInfo": {"context1": "info1", "context2": "info2"},
+    }
+]
+PD_DATA = DataFrame([DICT_DATA])
+GEOMETRY = [Point(4, 2)]
+GEOPD_DATA = GeoDataFrame([DICT_DATA], crs="EPSG:4326", geometry=GEOMETRY)
```

Comment:
Are we using those?

Reply:
Actually no. Will use in the next steps.
I will remove for now.

Reply:
Ok, as a reviewer I'll need more clarity on these next steps and on this PR:
* what is the purpose of this PR
* what is the purpose of next PRs

Reply:
The purpose of this PR is to create the flatten function for the crossfire data requested as dict;
My idea is to expand this function so the flatten function can work on `DataFrame` (I believe that does'nt matter if `dataFrame` or `GeoDataFrame`) cross fire data. 

I thoughg of doing this in a different PR. But I coul change the name of this one and start working on the next step;
What would you suggest?

Reply:
> What about flattening a list of dictionaries? Is it part of this PR or of a follow-up PR?

Its done.
Sorry the last commit I didn't commented the changes in the code.
To confirm, I have ran:
```python
>>> from crossfire import NestedColumnError, flatten
>>> data = [
    {
        "answer": 42,
        "contextInfo": {"context1": "info1", "context2": "info2"},
    }, {
        "answer": 51,
        "contextInfo": {"context1": "info2", "context2": "info3"},
    }
]
>>> flatten(data)
[{'answer': 42, 'contextInfo_context1': 'info1', 'contextInfo_context2': 'info2'}, {'answer': 51, 'contextInfo_context1': 'info2', 'contextInfo_context2': 'info3'}]
```

Reply:
It is nice to state that (what this PR should cover and what are next steps) in the PR description.

Reply:
Thanks I will consider in the next PR.